### PR TITLE
Restructure expiry functions

### PR DESCRIPTION
**Pull request by katomat**
Goal:
Have a function for other scripts to anonymise individual contacts.
Add a little_helpers service to anonymise a contact.
Expiry Cron jobs should call it instead of providing this functionality themselves.

### CODE REVIEW

**Comment by katomat** [comment]
[Campaignion: Expiry](https://trello.com/c/EOHDgG5v/1316-campaignion-expiry)

File: campaignion_expiry/src/Anonymizer.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+/**
+ * Anonymization service for redhen contacts.
+ */
+class Anonymizer {
+
+  /**
+   * Anonymize a single contact.
+   *
+   * @param \Drupal\campaignion\Contact $contact
+   *   Contact that is being anonymized.
```
Comment (torotil) [inline campaignion_expiry/src/Anonymizer.php]: The `Anonymizer` gets the contact passed as argument to the method call not to the constructor. That would also match the service definition.

```suggestion
```

File: campaignion_expiry/tests/AnonymizeContactTest.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+use Drupal\campaignion\Contact;
+
+/**
+ * Test anonymizing a contact.
+ */
+class AnonymizerTest extends DrupalUnitTestCase {
+
+  /**
+   * Set up.
+   */
```
Comment (torotil) [inline campaignion_expiry/tests/AnonymizeContactTest.php]: A more informative comment would be nice (also below at `tearDown()`.

File: campaignion_expiry/tests/AnonymizeContactTest.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  /**
+   * Test anonymizing our contact.
+   */
+  public function testContactAnonymization() {
+    $anonymizer = new Anonymizer(false);
+    $anonymizer->anonymize($this->contact, false);
+    $contact = entity_load_single('redhen_contact', $this->contact->contact_id);
+    $this->assertNotEmpty($contact);
+    $this->assertEqual($contact->first_name, "Anonymous");
+  }
```
Comment (torotil) [inline campaignion_expiry/tests/AnonymizeContactTest.php:52]: We could/should also test `Anonymizer::deleteOldRevisions()` here.

**Comment by katomat** [comment]
I added more comments and added testing for the deleteOldRevisions functions.

File: campaignion_expiry/tests/AnonymizeContactTest.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Drupal\campaignion_expiry;
+
+use Drupal\campaignion\CRM\Import\Source\ArraySource;
+use Drupal\campaignion\ContactTypeManager;
+use Drupal\campaignion_expiry\Anonymizer;
+use Upal\DrupalUnitTestCase;
+use Drupal\campaignion\Contact;
```
Comment (torotil) [inline campaignion_expiry/tests/AnonymizeContactTest.php]: Could you sort the use statements alphabetically?

File: campaignion_expiry/tests/AnonymizeContactTest.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+/**
+ * Test anonymizing a contact.
+ */
+class AnonymizerTest extends DrupalUnitTestCase {
+
+  /**
+   * Set up.
+   *
```
Comment (torotil) [inline campaignion_expiry/tests/AnonymizeContactTest.php]: No need to keep the worthless part of the comment. ;-)

**Review by torotil — CHANGES_REQUESTED** [review]
… otherwise good enough for me :-)

**Comment by katomat** [comment]
@torotil Wouldn't the tearDown actually require to not only delete the $this->contact, but also the actual anonymised redhen_contact?

**Comment by torotil** [comment]
> @torotil Wouldn't the tearDown actually require to not only delete the $this->contact, but also the actual anonymised redhen_contact?

They are the same — as in having the same `$contact->contact_id` that’s what anonymization means after all.

**Review by torotil — APPROVED** [review]

**Pull request merged**